kasa/experimental/modules/device.py

```diff
@@ -0,0 +1,40 @@
+"""Implementation of device module."""
+
+from __future__ import annotations
+
+from ...feature import Feature
+from ..smartcameramodule import SmartCameraModule
+
+
+class DeviceModule(SmartCameraModule):
+    """Implementation of device module."""
+
+    NAME = "devicemodule"
+    QUERY_GETTER_NAME = "getDeviceInfo"
+    QUERY_MODULE_NAME = "device_info"
+    QUERY_SECTION_NAMES = ["basic_info", "info"]
+
+    def _initialize_features(self):
+        """Initialize features after the initial update."""
+        self._add_feature(
+            Feature(
+                self,
+                id="device_id",
+                name="Device ID",
+                attribute_getter="device_id",
+                category=Feature.Category.Debug,
+                type=Feature.Type.Sensor,
+            )
+        )
+
+    async def _post_update_hook(self):
+        """Perform actions after a device update.
+
+        Overrides the default behaviour to disable a module if the query returns
+        an error because this module is critical.
+        """
```

Comment:
This is noop?

Reply:
The default post update hook tries to access the data and will eventually disable the module after several errors. For the device module we replace this with a noop as we don’t want to disable the device module.

Reply:
I'd then replace the summary line stating that to make it clear that it's intentional, something like "Overridden to foobar".

Reply:
Isn't this docstring sufficient:
```
Overrides the default behaviour to disable a module if the query returns
an error because this module is critical.
```

Reply:
Well, sort of, but oftentimes it's better if the summary line doesn't contradict what's happening ;-) "perform actions" and then there's confusingly no method body at all.